backend/app/models/tutoring.rb

```diff
@@ -40,6 +40,24 @@ class Tutoring < ApplicationRecord
   # Futuras (scheduled_at >= ahora)
   scope :upcoming, -> { where(scheduled_at: Time.current..) }
 
+  scope :search_by_course_name, ->(q) {
+    return all if q.blank?
+
+    joins(:course).where("courses.name ILIKE ?", "%#{q}%")
```

Comment:
Para que sea igual que en subjects, que ignore los acentos

```suggestion
   joins(:course).where("unaccent(courses.name) ILIKE unaccent(?)", "%#{q}%")
```